core/src/main/java/org/apache/iceberg/avro/AvroSchemaUtil.java

```diff
@@ -347,6 +347,13 @@ static Schema copyRecord(Schema record, List<Schema.Field> newFields, String new
     return copy;
   }
 
+  public static String makeCompatibleName(String name) {
```

Comment:
Nit: do you want to use this method in org.apache.iceberg.avro.TypeToSchema#struct() api? 
Currently the code is 
```
boolean isValidFieldName = AvroSchemaUtil.validAvroName(origFieldName);
String fieldName =  isValidFieldName ? origFieldName : AvroSchemaUtil.sanitize(origFieldName);
```

Reply:
I considered it, but the `isValidFieldName` is reused to add the original name as a property, so I think it's fine as it is.